site/templates/message.html

```diff
@@ -53,122 +56,134 @@ <h2>Send Brian a message on his Tidbyt.</h2>
+    sending: "Sending...",
+  };
+
+  const toggleButton = (name) => {
+    elements[name].button.toggleAttribute("disabled");
+    const isSending = elements[name].text.innerHTML === strings.sending;
+    elements[name].text.innerHTML = isSending ? strings[name] : strings.sending;
+  };
+
+  const setResponseText = (name, text) => {
+    elements[name].response.innerHTML = text;
+    // clear response message after 5 seconds
+    setTimeout(() => (elements[name].response.innerHTML = ""), 5000);
+  };
 
-  //sends the stop api request and displays the returned text
+  // sends the stop api request and displays the returned text
   const sendStop = async () => {
-    toggleButton("stop-button", "stop-button-span", "Tell Brian to stop talking.", "Sending...")
-    const response = await fetch("https://api.brianwtracey.com/stop");
-    const myJson = await response.json();
-    const responseDiv = document.getElementById("response_messages");
-    removeAllChildNodes(responseDiv);
-    const newContent = document.createTextNode(myJson);
-    responseDiv.appendChild(newContent);
-
-    toggleButton("stop-button", "stop-button-span", "Tell Brian to stop talking.", "Sending...")
-    // clear response message after 5 seconds
-    setTimeout(() => {removeAllChildNodes(responseDiv)}, 5000);
-  }
+    toggleButton("stop");
 
-  // add form submit listener with custom submission logic
-  const form = document.getElementById("message-form");
-  form.addEventListener("submit", async (event) => {
+    const text = await api.stop();
+    setResponseText("stop", text);
+
+    toggleButton("stop");
+  };
+
+  const sendMessage = async (event) => {
     event.preventDefault();
 
-    await toggleButton("submit-button", "submit-button-span", "Send Message", "Sending...");
+    toggleButton("message");
 
     const formData = new FormData(event.target);
     const formJSON = Object.fromEntries(formData.entries());
-    const resp = await fetch("https://api.brianwtracey.com/message", {
-      method: 'post',
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(formJSON)
-    });
-    const respJson = await resp.json();
-    const responseDiv = document.getElementById("message-response-div");
-    const responseMessage = respJson?.status == "success" ? "Message sent!" : "Failed to send message.";
-    const newContent = document.createTextNode(responseMessage);
-    removeAllChildNodes(responseDiv);
-    responseDiv.appendChild(newContent);
+    const response = await api.message(formJSON);
+    const text = response?.status === "success" ? "Message sent!" : "Failed to send message.";
+    setResponseText("message", text);
     form.reset();
 
-    await toggleButton("submit-button", "submit-button-span", "Send Message", "Sending...");
-
-    // clear response message after 5 seconds
-    setTimeout(() => {removeAllChildNodes(responseDiv)}, 5000);
-  });
+    toggleButton("message");
+  };
```

Comment:
With all the other logic encapsulated, `sendStop` and `sendMessage` become much easier to read and see what the whole thing is doing.

I abstracted `sendMessage` to its own function for parity with `sendStop`. It is still attached to the form below.